core/lib/Cardano/Address/Script.hs

```diff
@@ -462,10 +467,18 @@ validateScriptTemplate level (ScriptTemplate cosigners' script) = do
     let unusedCosigners =
             Set.fromList (Map.keys cosigners') `difference` allCosigners
     unless (Set.null unusedCosigners) $ Left UnusedCosigner
-    first WrongScript $ do
-        requiredValidation script
-        when (level == RecommendedValidation ) $
-            first NotRecommended (recommendedValidation script)
+
+-- | Validate a 'ScriptTemplate', semantically
+--
+-- @since 3.5.0
+validateScriptInScriptTemplate
+    :: ValidationLevel
+    -> ScriptTemplate
+    -> Either ErrValidateScript ()
+validateScriptInScriptTemplate level (ScriptTemplate _ script) = do
```

Comment:
I think the `ScriptTemplate` arg could be `Script KeyHash` by itself.
Can `validateScript` be used instead of this new function?